### ISSUE TITLE
[themes] Latest update makes syntax highlighting too intensive

### ISSUE THREAD

**Issue by Ratismal** [Microsoft/vscode#15039]
- VSCode Version: Code 1.7.1 (02611b40b24c9df2726ad8b33f5ef5f67ac30b44, 2016-11-03T13:53:14.822Z)
- OS Version: Windows_NT ia32 10.0.14393
- Extensions:

|Extension|Author|Version|
|---|---|---|
|vscode-theme-onedark|akamud|0.2.10|
|theme-onedark|azemoh|0.3.2|
|vscode-eslint|dbaeumer|1.0.8|
|beautify|HookyQR|0.5.4|
|csharp|ms-vscode|1.4.1|

---

One of my favourite parts of VSCode is the unobtrusive syntax highlighting. The only things that were highlighted were what was needed, making code very easy to read.

However, now it seems VSCode is trying to copy other IDEs such as Atom and Sublime Text, throwing colour absolutely everywhere. This makes reading my own code incredibly confusing, as I don't know what is what anymore.

Example:
![image](https://cloud.githubusercontent.com/assets/1059362/20030877/839f3e12-a333-11e6-9434-357c738ed7d6.png)

Is there a way to revert to the old syntax highlighting scheme?

**Comment by TheColorRed** [comment]
They should go back for people who like the simple color scheme, and create a more vibrant color scheme for those who do like the color.

**Comment by bgse** [comment]
For the two main (Dark and Light) themes, there are two variants each, e.g. "Dark+" will be quite colorful, and "Dark" just color the bare minimum.

The default is the colorful "Dark+" though.

**Comment by Ratismal** [comment]
It's not the same though. While it's less obtrusive, it still uses the new token system. I'd like to see the option to switch between the new and old token systems, as this new update affects every single theme.

**Comment by skid** [comment]
I use the solarized syntax theme, which is one of the default themes. After the latest update the methods, properties, variables and function arguments are all suddenly blue. Having a means to at least change the color of different tokens would be very helpful.

**Comment by rebornix** [comment]
This one shares the same reason as #14921 .

**Comment by infinnie** [comment]
@rebornix you mean it wasn’t intended to be highlighted like this?

@Ratismal you can currently use the themes Dark (Visual Studio)/Light (Visual Studio) for an unobtrusive syntax highlighting.

**Comment by Ratismal** [comment]
@infinnie As I said previously, this update still affects every existing theme. Also mentioned in the #14921, this update does things such as making literal templates being the exact same colour as regular strings. Using the minimal Dark/Light themes are not a solution.

**Comment by linusbrolin** [comment]
I completely agree that the minimal Dark/Light themes are way too minimal.
I liked the Dark+ theme, because it added _some_ more syntax highlighting while still being unobtrusive.
But now, everything is light blue or light yellow or brown.

Replacing almost everything that was uncolored (white) with light blue is not an improvement.
And I've tried recreating the Dark+ theme as well as possible, but there doesn't seem to be any way to do it anymore.
It's either _everything_ light blue, or _nothing_ light blue, which is not a solution.
Same with light yellow.
And there are also some problems with some things getting wrongly colored green.
I've tried to explain the problem here: https://github.com/Microsoft/vscode/issues/15001

Basically, the `support.class` scope colors too many things green that it shouldn't.
And the `variable` scope colors _way_ too many things light blue, making everything light blue.

What if I want only the variables to be light blue, and any keys/parameters within those variables be white? That doesn't seem to be possible anymore.

Right now, it's either everything or nothing, and that's a big step backwards in my opinion.

**Comment by thorn0** [comment]
As for the Solarized theme, for comparison, here is what it's supposed to look like. This screenshot is from [the site](http://ethanschoonover.com/solarized) of the Solarized project.

![image](https://cloud.githubusercontent.com/assets/94334/20153888/9272373a-a6cd-11e6-9489-3fae35ba1b68.png)

And VS Code (today's Insiders version):

![image](https://cloud.githubusercontent.com/assets/94334/20153881/81cc620c-a6cd-11e6-87e9-7138d92c3a67.png)

Please see the [Usage & Development](http://ethanschoonover.com/solarized#usage-development) part of the site. According to the design of this palette, for the optimal contrast, the main color of text on dark background is supposed to be `base00` (gray, `#657b83`) whereas now in VS Code, text is mostly blue.

**Comment by Ratismal** [comment]
As an update, I've found a workaround where I took the old javascript grammar from [this commit](https://github.com/Microsoft/vscode/tree/5ca799fd0729cec5e7c30e70022f37bb3cb4b608/extensions/javascript) and inserted it into vscode as a new extension. While, I'd prefer a native way to revert, this will do for now.

Here's the same code as I uploaded before, but with the legacy grammar:
![image](https://cloud.githubusercontent.com/assets/1059362/20194092/2e87ff3a-a74e-11e6-8aad-737f1c62f368.png)

**Comment by linusbrolin** [comment]
@Ratismal Can you make that extension available for everyone so we have a solution until this problem is fixed?

**Comment by Ratismal** [comment]
Not sure if I've done it correctly, but I published it [here](https://marketplace.visualstudio.com/items?itemName=ratismal.legacy-js-grammar) @linusbrolin

**Comment by linusbrolin** [comment]
@Ratismal Thanks!
I've installed it but don't see any difference.
Do I need to change any settings or replace any themes aswell, or should it just work out of the box with the built in Dark+ theme?

**Comment by Ratismal** [comment]
@linusbrolin I'm not sure; simply having it installed and reloading worked for me.

I'm not very knowledgeable about making extensions, so I made it overwrite the default languages. This may do weird things to your instance. I think there's a way to register it as a new language so it doesn't overwrite, but I have no idea how to do that. I've uploaded the source [here](https://github.com/Ratismal/vscode-legacy-js-grammar) if anybody wants to take a look?

**Comment by aeschli** [comment]
We understand that theme changes are always controversial. But we had a lot of requests and complaints about not highlighting type and function references. We ended up with a full rewrite of the typescript grammar as well as updated Plus themes. The grammar rewrite did not only add the new scopes but also fix a lot of issues in the grammar.
We don't want to add two more 'default' themes, nor do we want to modify the Visual Studio themes which are based on the way that Visual Studio colorizes. An extension is the way to go forward. I'm happy to help.
@Ratismal I will comment in https://github.com/Ratismal/vscode-legacy-js-grammar
